### PR TITLE
add  keras2 GlobalMaxPooling2D/3D&GlobalAveragePooling3D layers and test

### PR DESCRIPTION
**Pull request by yjjiang11**

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling3D.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import scala.reflect.ClassTag
+
+/**
+ * Applies global max pooling operation for spatial data.
+ * The input of this layer should be 4D.
+ *
+ * When you use this layer as the first layer of a model, you need to provide the argument
+ * inputShape (a Single Shape, does not include the batch dimension).
+ *
+ * @param dataFormat Format of input data. Please use DataFormat.NCHW (dataFormat='channels_first')
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling3D.scala]: wrong doc...

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling3D.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Shape
+import com.intel.analytics.zoo.pipeline.api.Net
+import com.intel.analytics.zoo.pipeline.api.keras2.layers.utils.KerasUtils
+import com.intel.analytics.zoo.pipeline.api.keras.{layers => klayers1}
+import scala.reflect.ClassTag
+
+/**
+ * Applies global max pooling operation for spatial data.
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling3D.scala]: wrong doc...

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalMaxPooling3D.scala [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import com.intel.analytics.bigdl.utils.Shape
+import com.intel.analytics.zoo.pipeline.api.Net
+import com.intel.analytics.zoo.pipeline.api.keras2.layers.utils.KerasUtils
+import com.intel.analytics.zoo.pipeline.api.keras.{layers => klayers1}
+import scala.reflect.ClassTag
+
+/**
+ * Applies global max pooling operation for 3D data.
+ * Data format currently supported for this layer is 'CHANNELS_FIRST' (.
+ * Border mode currently supported for this layer is 'valid'.
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalMaxPooling3D.scala]: padding?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling3D.scala [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import com.intel.analytics.bigdl.utils.Shape
+import com.intel.analytics.zoo.pipeline.api.Net
+import com.intel.analytics.zoo.pipeline.api.keras2.layers.utils.KerasUtils
+import com.intel.analytics.zoo.pipeline.api.keras.{layers => klayers1}
+import scala.reflect.ClassTag
+
+/**
+ * Applies global average pooling operation for 3D data.
+ * Data format currently supported for this layer is 'CHANNELS_FIRST' .
+ * Border mode currently supported for this layer is 'valid'.
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling3D.scala]: remove??

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling3D.scala [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Shape
+import com.intel.analytics.zoo.pipeline.api.Net
+import com.intel.analytics.zoo.pipeline.api.keras2.layers.utils.KerasUtils
+import com.intel.analytics.zoo.pipeline.api.keras.{layers => klayers1}
+import scala.reflect.ClassTag
+
+/**
+ * Applies global average pooling operation for 3D data.
+ * Data format currently supported for this layer is 'CHANNELS_FIRST' .
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling3D.scala]: use lower case "channels_last" same as keras?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalMaxPooling3D.scala [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Shape
+import com.intel.analytics.zoo.pipeline.api.Net
+import com.intel.analytics.zoo.pipeline.api.keras2.layers.utils.KerasUtils
+import com.intel.analytics.zoo.pipeline.api.keras.{layers => klayers1}
+import scala.reflect.ClassTag
+
+/**
+ * Applies global max pooling operation for 3D data.
+ * Data format currently supported for this layer is 'CHANNELS_FIRST' (.
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalMaxPooling3D.scala]: lower case. remove (

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalMaxPooling3D.scala [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+ * padding currently supported for this layer is 'valid'.
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalMaxPooling3D.scala]: Padding

**Comment by hkvision** [comment]
Some minor doc change. Other LGTM
